File: Downloads/redaapp/react_app/src/services/ClientService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/clients';

const getClients = async () => {
    return await axios.get(API_URL);
};

const getClient = async (id) => {
    return await axios.get(`${API_URL}/${id}`);
};

export const addClient = async (client) => {
    return await axios.post(API_URL, client);
};

const deleteClient = async (id) => {
    return await axios.delete(`${API_URL}/${id}`);
};

const updateClient = async (client) => {
    return await axios.put(`${API_URL}/${client.id}`, client);
};

export const ClientService = {
    getClients,
    getClient,
    addClient,
    deleteClient,
    updateClient
};

File: Downloads/redaapp/src/main/java/com/example/ClientAzure/Service/ClientService.java
package com.example.ClientAzure.Service;

import com.example.ClientAzure.Bean.Client;

import java.util.List;

public interface ClientService {
    List<Client> getClients();
    Client getClient(long id);
    Client addClient(Client client);
    void deleteClient(long id);
    void updateClient(Client client);
}
